src/script/common/c_internal.cpp

```diff
@@ -46,6 +47,25 @@ int script_exception_wrapper(lua_State *L, lua_CFunction f)
 	return lua_error(L);  // Rethrow as a Lua error.
 }
 
+int script_error_handler(lua_State *L)
+{
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "error_handler");
+	if (!lua_isnil(L, -1)) {
+		lua_pushvalue(L, 1);
+	} else {
+		// No Lua error handler available. Call debug.traceback(tostring(#1), level).
+		lua_getglobal(L, "debug");
+		lua_getfield(L, -1, "traceback");
+		lua_getglobal(L, "tostring");
+		lua_pushvalue(L, 1);
+		lua_call(L, 1, 1);
```

Comment:
I think this would put the values in the wrong order on the stack. `debug.traceback` should be below the result of the `tostring` call, since `debug.traceback` will be called with the `tostring` result as its first argument.